### PR TITLE
Add implementation details for "Future REST API"

### PR DESCRIPTION
**Pull request by shannynalayna**
This change adds implementation details for the future Grid REST API to
the community planning section of the site.

Signed-off-by: Shannyn Telander <telander@bitwise.io>

### CODE REVIEW

File: community/planning/rest_api/index.md [hunk 1]
```diff
@@ -13,21 +13,174 @@ implemented as project requirements evolve.
...
+be circuit- or network-wide.
+
+### Payload submission
+
+Batches submitted to either Griddle or the Grid daemon will be handled in
+similar ways. The updated REST API will offer endpoints that will accept
+incoming batches as either binary data or JSON, and singular or multiples of
+batches. Specific contract actions will be reflected in the method and
+endpoint. For example, an endpoint used to update a location will be
+`PUT / locations/{location_id}`. An example of the Rust API for this endpoint
```
Comment (dnewh) [inline community/planning/rest_api/index.md:109]: extra space between the first `/` and `locations`

**Review by dnewh — CHANGES_REQUESTED** [review]

File: community/planning/rest_api/index.md [hunk 2]
```diff
@@ -13,21 +13,174 @@ implemented as project requirements evolve.
...
+Grid’s protobuf transactions. The REST API will also offer functionality to
+create and sign transactions and batches, depending on the enabled features and
+which component is being accessed.
+
+This implementation uses Actix Web 3.0, further documentation can be found
+here: https://actix.rs/.
+
+### Griddle
+
+Griddle has two primary functions, signing and proxying requests to the Grid
```
Comment (CLEckhardt) [inline community/planning/rest_api/index.md:51]: , should be :

File: community/planning/rest_api/index.md [hunk 3]
```diff
@@ -13,21 +13,174 @@ implemented as project requirements evolve.
...
+transactions, for example when using the `/batches` endpoint. The `target`
+refers to the endpoint resource the batch is structured after and determines
+how the payload is deserialized. The rest of the values in the payload
+resources are transaction-specific.
+
+### Request Headers
+
+When creating a new batch, users may want to set their own identifier for the
+state changes. This ID may correspond to external systems or data that inform
+the batch. An `event_id` may be set for a batch and can be used
```
Comment (CLEckhardt) [inline community/planning/rest_api/index.md:96]: `data_change_id`

**Review by CLEckhardt — APPROVED** [review]

**Comment by shannynalayna** [comment]
As our approach has shifted slightly, I will be closing this PR in favor of one to add more up-to-date information on our plan moving forward with the Grid REST API.

**Pull request closed without merging**